sql/src/test/java/org/apache/druid/quidem/DruidQuidemCommandHandler.java

```diff
@@ -141,25 +156,21 @@ protected void executeExplain(Context x) throws Exception
     {
       DruidConnectionExtras connectionExtras = (DruidConnectionExtras) x.connection();
       ObjectMapper objectMapper = connectionExtras.getObjectMapper();
-      QueryLogHook qlh = new QueryLogHook(objectMapper);
-      qlh.logQueriesForGlobal(
-          () -> {
-            executeQuery(x);
-          }
-      );
-
-      List<Query<?>> queries = qlh.getRecordedQueries();
-
-      queries = queries
-          .stream()
-          .map(q -> BaseCalciteQueryTest.recursivelyClearContext(q, objectMapper))
-          .collect(Collectors.toList());
+      DruidHookDispatcher dhp = unwrapDruidHookDispatcher(x);
```

Comment:
while I can follow whats happening here, would be nice to add some comments alongside the code.

Reply:
instead of adding a comment; I've extracted a method named `executeExplainCollectHookValues`;
I think that too increased readability and also consolidated to do the same from 3 callsites